oxidation/libparsec/crates/core_fs/src/storage/user_storage.rs

```diff
@@ -1,30 +1,50 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 
-use std::collections::HashSet;
+use std::{collections::HashSet, path::Path};
 
 use libparsec_client_types::{LocalDevice, LocalManifest, LocalUserManifest};
 use libparsec_types::EntryID;
 
-use super::manifest_storage::ManifestStorage;
+use super::{
+    local_database::SqliteConn, manifest_storage::ManifestStorage,
+    version::get_user_data_storage_db_path,
+};
 use crate::error::{FSError, FSResult};
 
 pub struct UserStorage {
-    device: LocalDevice,
-    user_manifest_id: EntryID,
+    pub device: LocalDevice,
+    pub user_manifest_id: EntryID,
     manifest_storage: ManifestStorage,
+    data_conn: SqliteConn,
 }
 
 impl UserStorage {
-    pub fn new(
+    pub fn from_db_dir(
```

Comment:
Why passing a directory instead of just the path of the db file ?

Reply:
1. When we create a new `UserStorage` with `UserStorage::run` (in python) we only have the said directory path
2. `get_{user,workspace}*_db_path` will be removed in this PR (currently not but it's a miss from my part)
    With that said the python code will not be able to provide the full path (up to the file)

This can change but would require to provide on oxidation of `get_{user,workspace}*_db_path`, this is up to you

Reply:
ok so let's stick to the simplest way then ;-)